frontend/src/components/stacks/ExpensesStack/ExpensesStack.jsx

```diff
@@ -139,36 +94,35 @@ export default function ExpensesDirectionStack() {
   }, [costAnalyticState, expensesState]);
 
   const onExpenseChange = (expense, event) => {
-    console.log("Event targe value: " + event.target.value);
     expense.value = event.target.value;
 
     const updatedExpenses = expenses.map((item) => {
       return item.id === expense.id ? expense : item;
     });
     let costAnalytic = costAnalyticState.getState();
-    costAnalytic.dailyRecommended -= 10;
+    costAnalytic.dailyRecommended -= 10; // TODO remove/fix this when dynamic update of daily recommended is fixed
     costAnalyticState.setState(costAnalytic);
     expensesState.setState(updatedExpenses);
   };
 
-  // edit expense
+  // Edit expense event
   const handleKeyDown = (expense, event) => {
     if (event.key === "Enter") {
       expense.value = event.target.value;
-      axios
-        .put(config.server.uri + "expenses/" + expense.id, expense, {
-          headers: {
-            "Content-Type": "application/json",
-          },
-        })
-        .then((response) => {
-          console.log("[ExpensesStack]: RESPONSE OK: " + response.data);
-          toast.success("Edited expense stored!");
-        })
-        .catch((error) => {
-          console.log("[ExpensesStack]: RESPONSE ERROR: " + error);
-          toast.error("Edited expense not stored!");
-        });
+      editExpense(expense);
+    }
+  };
+
+  const removeExpense = (expense, event) => {
+    console.log("[ExpensesStack]: Will delete item with id: " + expense.id);
+    deleteExpense(expense);
+
+    var array = [...expenses];
+    var index = array.indexOf(expense);
+    if (index !== -1) {
+      array.splice(index, 1);
+      setExpenses(array);
+      expensesState.setState(array);
```

Comment:
We don't actually need this. 
The entire business logic must be handled by the state
```suggestion
   expensesState.modifyExpense(expense);
```